pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -646,11 +628,9 @@ def _auto_select_batch_size(self):
                     )
         return batch_size
 
-    def _format_precision_config(self):
-        if self.amp_type == AMPType.APEX:
-            amp_level = self.amp_level
-        if self.precision in (16, "mixed"):
-            if "fp16" not in self.config and self.amp_type == AMPType.NATIVE:
+    def _format_precision_config(self) -> None:
+        if self.precision_plugin.precision in (16, "mixed"):
```

Comment:
should we compare `PrecisionType` here instead of the actual values?

Reply:
That's fine with me. If you don't mind, I would prefer doing it across all plugins in a separate PR #10704